### ISSUE TITLE
Candidate Tools to add

### ISSUE THREAD

**Issue by ozbillwang** [alpine-docker/k8s#38]
1. helmfile (https://github.com/helmfile/helmfile)
2. kubectx + kubens (https://github.com/ahmetb/kubectx)

**Comment by ozbillwang** [comment]
any interestings to put below tools as well?

* jsonnet (https://jsonnet.org/articles/kubernetes.html)
* kubecfg (https://github.com/kubecfg/kubecfg)
* make (do we really need make here?)

**Comment by ozbillwang** [comment]
Anything else?

**Comment by n4nn31355** [comment]
make actually will be appreciated because it helps to keep some jobs identical and so up to date both in CI and locally. For example, `helm diff\lint`, parameterized build commands, etc

**Comment by caleb-devops** [comment]
I'd like terraform if possible.

**Comment by dhduvall** [comment]
[dyff](https://github.com/homeport/dyff) would be nice. Could even set `KUBECTL_EXTERNAL_DIFF="dyff between --omit-header --set-exit-code"`.

**Comment by ozbillwang** [comment]
New tool to me, thanks. 

what's the different to [helm-diff](https://github.com/databus23/helm-diff)

**Comment by dhduvall** [comment]
It's a stand-alone tool instead of a plugin, so it operates on YAML files rather than running the `helm` command as necessary to extract that data. It does have some special knowledge about Kubernetes manifests, though, so it can do a bit better with them than it would if it treated them as generic YAML.

But what I think stands out about it is that it does a really good job at figuring out which items in each manifest to compare, and presenting them in a really intuitive fashion. The `helm-diff` README doesn't give a particularly clear example of what its output looks like, and I'm not going to figure it out tonight, but I'll give it a whirl tomorrow to see. Judging from the diffs in the test suite, though, I'd say that `dyff` gave up on trying to make the diffs look like the output of `diff` and instead made the diff sections follow the structure of the underlying files, and taking into account special things like reordering of lists.

**Comment by ozbillwang** [comment]
DockerFile lint :

Dockle
Hadolint

**Comment by ozbillwang** [comment]
security command recommends to put in Dockerfile.
```
RUN find / -xdev -perm /6000 -type f -exec chmod a-s {} \"; || true
```

**Comment by WesselAtWork** [comment]
I would like to see [perl-mime-construct](https://pkgs.alpinelinux.org/package/edge/community/x86_64/perl-mime-construct) to specifically make composing raw email documents easier for `aws ses send-raw-email` (which [only takes](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ses/send-raw-email.html) a raw MIME formatted email document as input)

I saw the we already have the python [email](https://docs.python.org/3/library/email.html) package installed so if anyone knows of a python alternative that provides does the same command line level functionality I wouldn't mind including that instead.

**Comment by ozbillwang** [comment]
@WesselAtWork 

Anything related to Kubernetes? For mail sending, should we setup another task in pipeline to do that seperately?

**Comment by coomdan** [comment]
I would like to have [kube-conform](https://github.com/jtyr/kubeconform-helm) plugin

**Comment by snup67** [comment]
it will be nice to add ansible so all helm command can run from ansible playbook

**Comment by ozbillwang** [comment]
> it will be nice to add ansible so all helm command can run from ansible playbook

thanks, I have replied in https://github.com/alpine-docker/k8s/issues/61#issuecomment-1626884684

**Comment by ozbillwang** [comment]
> I would like to have [kube-conform](https://github.com/jtyr/kubeconform-helm) plugin

sure, let me take a look

**Comment by ozbillwang** [comment]
> I would like to have [kube-conform](https://github.com/jtyr/kubeconform-helm) plugin

I am thinking if this plugin need be added into `alpine/k8s` or `alpine/helm`? Or both?

**Comment by coomdan** [comment]
> > I would like to have [kube-conform](https://github.com/jtyr/kubeconform-helm) plugin
> 
> I am thinking if this plugin need be added into `alpine/k8s` or `alpine/helm`? Or both?

definitively in `alpine/helm`, but would make sense in both

**Comment by ozbillwang** [comment]
> Hello,
> 
> It would be nice to have flux, thanks !

good idea, i am thinking to create another image includes Argocd and flux command line, because they are similar and not depend on k8s

**Comment by mklauber** [comment]
We use envsubst as a very lightweight templating tool.

**Comment by enflo** [comment]
Hello

Would it be possible to add krew?

Krew docker code:

```
# Installing Krew
RUN OS="$(uname | tr '[:upper:]' '[:lower:]')" && \
    ARCH="$(uname -m | sed -e 's/x86_64/amd64/' -e 's/\(arm\)\(64\)\?.*/\1\2/' -e 's/aarch64$/arm64/')" && \
    curl -fsSLO "https://github.com/kubernetes-sigs/krew/releases/latest/download/krew.tar.gz" && \
    tar zxvf krew.tar.gz && \
    KREW=./krew-"${OS}_${ARCH}" && \
    "$KREW" install krew && \
    cp $HOME/.krew/bin/kubectl-krew /usr/local/bin/
```

**Comment by ozbillwang** [comment]
sure, can you raise PR for it?

reference the codes 

[curl -fsSLO "https://github.com/kubernetes-sigs/krew/releases/latest/download/krew.tar.gz"](https://github.com/alpine-docker/k8s/blob/master/Dockerfile#L14-L23)

to set the `ARCH` variable

**Comment by enflo** [comment]
Hi @ozbillwang 

https://github.com/alpine-docker/k8s/pull/63

As you told me, I have opened a PR with the changes to add Krew, I have left a comment in the PR of a small problem that I did not know how to solve.

Greetings

**Comment by ozbillwang** [comment]
Thanks @enflo I have merged your PR. 

Until now, the image size has grown to 700+MB, I am thinking to freeze it and don't add more tools in it.
